crates/ruff/src/rules/flake8_annotations/rules.rs

```diff
@@ -17,20 +17,24 @@ use crate::visibility::Visibility;
 
 define_violation!(
     /// ## What it does
+    ///
```

Comment:
Is there any way not to enforce these empty lines? I feel like they're not super helpful especially when we show users the Markdown directly. Maybe I'm wrong though?

Reply:
There is no option for this. `markdownlint,` `mdformat,` and `prettier`all add the empty lines.
An option would be to only format the code blocks content with black.
But thinking about it now it would make more sense to run ruff auto fix and formatter on it.
We would need a way to specify a ruff configuration for each code blocks.
It would signal to the developer when generating the docs that the code blocks were updated and thus incorrect.
For example, when we have the before and after examples.

Reply:
How exactly is this being run today? Like does `mdformat` get run over these Rust files, and it picks up the Markdown?

Reply:
No, I generated the files then ran mdformat on them, then edited the rust files until the generation produced valid markdown.

Reply:
I see, so would these not be enforced on an ongoing basis?

I'm partial to not applying the empty lines. [Clippy](https://github.com/rust-lang/rust-clippy/blob/3d193fa17a7df3256d58b841a2df92af1cbdbea4/declare_clippy_lint/src/lib.rs#L100) doesn't do that, and I like to follow Clippy as a best practice.

Reply:
No, they would not be enforced.

Reply:
Can I back out the changes to add newlines everywhere, then merge this?

Reply:
I can do it if you want.